### DISCUSSION TITLE
The same PROMQL statement aggregation query return different results

### DISCUSSION THREAD

**Discussion by daicheng123** [prometheus/prometheus#9729] [Q&A]
![image](https://user-images.githubusercontent.com/42512975/141289108-9b86e9b2-1af1-406b-9ba0-2b008007eafb.png)
![image](https://user-images.githubusercontent.com/42512975/141289184-3a0120a7-385a-4bdd-9a81-77b84e71e4c4.png)

**Comment by GiedriusS** [comment]
This is expected because rate() does extrapolation and as time goes on, your steps might fall on different timestamps as in your screenshots giving different results :smiley: Something like `query-frontend` from Thanos/Cortex can do query alignment according to the step parameter so that you wouldn't see such "anomalies". I hope this helps.

**Comment by roidelapluie** [comment]
It is in general indeed recommended to have the range (`[1m]` in your promql query) be 3 or 4 times the scrape interval. You could try with eg. `rate([5m])`.

**Reply by roidelapluie** [reply]
The following talk provides more insights: https://www.youtube.com/watch?v=09bR9kJczKM
